contrib/src/main/java/com/datatorrent/contrib/avro/AvroToPojo.java

```diff
@@ -0,0 +1,428 @@
+              ((PojoUtils.Setter<Object, String>)afi.setterOrGetter).set(newObj,
+                  new String(tuple.get(afi.fieldInfo.getColumnName()).toString()));
+              break;
+
+            case LONG:
+              afi.setterOrGetter = PojoUtils.createSetterLong(getCls(), afi.fieldInfo.getPojoFieldExpression());
+              ((PojoUtils.SetterLong<Object>)afi.setterOrGetter).set(newObj,
+                  (long)tuple.get(afi.fieldInfo.getColumnName()));
+              break;
+
+            default:
+              afi.setterOrGetter = PojoUtils.createSetter(getCls(), afi.fieldInfo.getPojoFieldExpression(), Byte.class);
+              ((PojoUtils.Setter<Object, byte[]>)afi.setterOrGetter).set(newObj,
+                  serialize(tuple.get(afi.fieldInfo.getColumnName())));
+              break;
+
+          }
+
+        } catch (AvroRuntimeException | IOException e) {
+          LOG.error("Exception in setting value" + e.getMessage());
+          fieldErrorCnt++;
+        }
+
+      }
+    } catch (Exception ex) {
+      LOG.error("Generic Exception in setting value" + ex.getMessage());
+      errorCnt++;
+    }
+
+    return newObj;
+  }
+
+  /**
+   * Use reflection to generate field info values if the user has not provided
+   * the inputs mapping
+   */
+
+  public String generateFieldInfoInputs(Class<?> cls)
+  {
+
+    java.lang.reflect.Field[] fields = cls.getDeclaredFields();
+
+    StringBuilder sb = new StringBuilder();
+
+    for (int i = 0; i < fields.length; i++) {
+
+      java.lang.reflect.Field f = fields[i];
+      Class<?> c = ClassUtils.primitiveToWrapper(f.getType());
+      sb.append(f.getName() + FIELD_SEPARATOR + f.getName() + FIELD_SEPARATOR + c.getSimpleName().toUpperCase()
+          + RECORD_SEPARATOR);
+
+    }
+
+    return sb.substring(0, sb.length() - 1).toString();
+
+  }
+
+  public List<FieldInfo> createFieldInfoMap(String str)
+  {
+    fieldInfos = new ArrayList<FieldInfo>();
+
+    StringTokenizer strtok = new StringTokenizer(str, RECORD_SEPARATOR);
+
+    while (strtok.hasMoreTokens()) {
+      String[] token = strtok.nextToken().split(FIELD_SEPARATOR);
+
+      fieldInfos.add(new FieldInfo(token[0], token[1], SupportType.valueOf(token[2])));
+    }
+
+    return fieldInfos;
+  }
+
+  @OutputPortFieldAnnotation(schemaRequired = true)
+  public final transient DefaultOutputPort<Object> output = new DefaultOutputPort<Object>()
+  {
+    public void setup(PortContext context)
+    {
+      cls = context.getValue(Context.PortContext.TUPLE_CLASS);
+    }
+  };
+
+  @Override
+  public void setup(OperatorContext context)
+  {
+
+    columnFieldSetters = Lists.newArrayList();
+
+    try {
+      parseSchema();
+    } catch (IOException e) {
+      LOG.error("Exception in parsing schema");
+    }
+
+    if (getFieldInfos().isEmpty() || getGenericRecordToPOJOFieldsMapping() == null) {
+      setFieldInfos(createFieldInfoMap(generateFieldInfoInputs(getCls())));
```

Comment:
Is this working?

cls variable is set in output Port's setup method which gets called after this method. So getCls would return null here?